### ISSUE TITLE
False negative validation error on duplicate job with no steps

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#224]
### Version

ghlint 0.5.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
```yaml

  a-job:
    name: "Job"
    runs-on: ubuntu-latest
    steps:

  a-job:
    name: "Job"
    runs-on: ubuntu-latest
    steps:

      - name: "Checkout ${{ github.ref }} in ${{ github.repository }} repository."
        uses: actions/checkout@v4
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

 * steps: needs 1 item: https://github.com/SchemaStore/schemastore/blob/29d7f160aa019c22fbb34602b554f891921d1b50/src/schemas/json/github-workflow.json#L773
 * duplicate job id

### Actual behavior

nothing reported

### Additional Context

a-7277

**Comment by TWiStErRob** [comment]
Duplicate job id is now reported via #273, the 1-item-step-array problem was hidden because of the duplicate key picking up the job later defined in the same workflow. Fixing the duplicate will lead to showing more problems.
